lib/search.rb

```diff
@@ -542,20 +542,36 @@ def post_action_type_filter(posts, post_action_type)
 
   advanced_filter(/\Awith:images\z/i) { |posts| posts.where("posts.image_upload_id IS NOT NULL") }
 
-  advanced_filter(/\Acategory:(.+)\z/i) do |posts, match|
+  advanced_filter(/\Acategory:(.+)\z/i) do |posts, terms|
     exact = false
 
-    if match[0] == "="
-      exact = true
-      match = match[1..-1]
+    matches = terms.split(",")
+
+    category_ids_found = []
+
+    matches.each do |match|
+      if match[0] == "="
+        exact = true
+        match = match[1..-1]
+      end
+
+      category_ids =
+        Category.where("slug ilike ? OR name ilike ? OR id = ?", match, match, match.to_i).pluck(
+          :id,
+        )
```

Comment:
This is going to result in one SQL query per category name. I think we can just collect all the category names first and then do something like `ILIKE ANY (ARRAY(...))` so that we only execute one query to get the ids from the database.

Reply:
The not exact default loophole is hard to implement though in multi category scenario :(